core/src/runtime/mod.rs

```diff
@@ -509,6 +522,17 @@ impl Runtime {
 
 impl Drop for Runtime {
     fn drop(&mut self) {
+        // the default rust drop order is self before fields, which is the wrong
+        // order for what EAL needs. so we use a trick here to take ownership of
+        // the inner fields and make them drop first.
+        self.ports.take();
```

Comment:
brought up the idea of using [manuallydrop](https://doc.rust-lang.org/std/mem/struct.ManuallyDrop.html) here instead of the options + take approach.